### PR TITLE
feat: Save reports to cluster, add new `--print` flag to print them as JSON

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Fix https://github.com/kubewarden/audit-scanner/issues/36

Save all reports to cluster.
Add a new `--print` flag: these flag prints the results to stdout as JSON, in addition to what the audit-scanner is doing.

## Test

Tested manually against cluster, see CONTRIBUTING.md.

### Potential improvement

This PR could include the first e2e tests for audit scanner (mocking policy-server). We also have k8s e2e tests listed as a different task on the epic.

### CODE REVIEW

**Review by jvanz — CHANGES_REQUESTED** [review]

File: internal/report/store.go [hunk 1]
```diff
@@ -156,9 +157,26 @@ func (s *PolicyReportStore) ToJSON() (string, error) {
...
+		if err := s.savePolicyReport(&report); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// savePolicyReport instantiates the passed namespaced PolicyReport if it doesn't exist, or
+// updates it if one is found
+func (s *PolicyReportStore) savePolicyReport(report *PolicyReport) error {
```
[the remark this answers is not in the document]
Comment (viccuad) [reply internal/report/store.go]: true! done in https://github.com/kubewarden/audit-scanner/pull/42/commits/d6e7550e9146a17a0f7b1c10e8c3a8e80d232ad0.

File: internal/report/store.go [hunk 2]
```diff
@@ -199,11 +218,64 @@ func (s *PolicyReportStore) Save(report *PolicyReport) error {
...
+					Str("report name", report.Name).Str("report ns", report.Namespace),
+				).Msg("updated PolicyReport")
+		}
+	}
+	return nil
+}
+
+// SavePolicyClusterPolicyReport instantiates the ClusterPolicyReport if it doesn't exist, or
+// updates it one is found
+func (s *PolicyReportStore) saveClusterPolicyReport() error {
```
[the remark this answers is not in the document]
Comment (viccuad) [reply internal/report/store.go]: true! done in https://github.com/kubewarden/audit-scanner/pull/42/commits/d6e7550e9146a17a0f7b1c10e8c3a8e80d232ad0.

File: internal/report/store.go [hunk 3]
```diff
@@ -199,11 +218,64 @@ func (s *PolicyReportStore) Save(report *PolicyReport) error {
 				Dict("dict", zerolog.Dict().
 					Str("report name", report.Name).Str("report ns", report.Namespace),
 				).Msg("PolicyReport update failed")
+			// TODO bubble up error
+			//
```
[the remark this answers is not in the document]
Comment (viccuad) [reply internal/report/store.go]: I have refactored the error handling of SaveAll() and the other 2 Save* functions, see https://github.com/kubewarden/audit-scanner/pull/42/commits/957144c86503b664353e633e3993f74f3d73397b.
Now they bubble up errors correctly.

**Review by flavio — CHANGES_REQUESTED** [review]

File: internal/report/store.go [hunk 4]
```diff
@@ -156,9 +157,29 @@ func (s *PolicyReportStore) ToJSON() (string, error) {
...
 
-// Save instantiates the passed namespaced PolicyReport if it doesn't exist, or
-// updated a new one if one is found
-func (s *PolicyReportStore) Save(report *PolicyReport) error {
+// SaveAll saves the store policy reports to the cluster. It does this by
+// instantiating new clusterPolicyReport and PolicyReports, or updating them if
+// they are already present.
+func (s *PolicyReportStore) SaveAll() {
+	if err := s.SaveClusterPolicyReport(); err != nil {
+		log.Error().Err(err).
```
[the remark this answers is not in the document]
Comment (viccuad) [reply internal/report/store.go:166]: Done in https://github.com/kubewarden/audit-scanner/pull/42/commits/ad050f45c91be6f9cd87801f030f00800bb57453!

**Review by flavio — APPROVED** [review]
Thanks for having submitted the changes requested :heart:

**Pull request merged**